src/routes/subscriptions.rs

```diff
@@ -32,59 +32,126 @@ impl TryFrom<FormData> for NewSubscriber {
     }
 }
 
+pub enum SubscribeError {
+    Validation(String),
+    StoreToken(StoreTokenError),
+    SendEmail(reqwest::Error),
+    Pool(sqlx::Error),
+    InsertSubscriber(sqlx::Error),
+    TransactionCommit(sqlx::Error),
+}
+
+impl std::error::Error for SubscribeError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            SubscribeError::StoreToken(e) => Some(e),
+            SubscribeError::SendEmail(e) => Some(e),
+            SubscribeError::InsertSubscriber(e) => Some(e),
+            SubscribeError::Pool(e) => Some(e),
+            SubscribeError::TransactionCommit(e) => Some(e),
+            SubscribeError::Validation(_) => None,
```

Comment:
`&str` doesnt implement `std::error::Error` so were just returning `None` for its source